src/LinearAlgebra.jl

```diff
@@ -392,15 +392,13 @@ control over the factorization of `A`.
 
 # Examples
 ```jldoctest
-julia> A = [1 2.2 4; 3.1 0.2 3; 4 1 2];
+julia> A, B = [1 2.2 4; 3.1 0.2 3; 4 1 2], [1, 2.5, 3];
 
-julia> X = [1; 2.5; 3];
+julia> Y = copy(B);
 
-julia> Y = zero(X);
+julia> ldiv!(Y, qr(A), B); # also try qr!(A) for the 2nd arg
```

Comment:
Probably the comment may explain the need for `qr!`, that is to reduce the allocations in `qr`.